File: PRACTICAS/PRACTICA_ACTUAL/RUBY/pkg/Civitas-0.0.1/lib/casilla_calle.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative 'casilla.rb'
require_relative 'titulo_propiedad.rb'
require_relative 'jugador.rb'

module Civitas
  class Casilla_calle < Casilla
  
      attr_reader :titulo_propiedad

     def initialize(titulo)
        super(titulo.nombre)
        @titulo_propiedad = titulo
      end
      
      #Override
      def recibe_jugador(iactual,todos)
       if (jugador_correcto(iactual,todos))
          informe(iactual,todos)
         jugador = todos[iactual]
         if !@titulo_propiedad.tiene_propietario
           jugador.puede_comprar_casilla
         else
           @titulo_propiedad.tramitar_alquiler(jugador)
         end
       end
     end
     
    #Override
    def to_s

      puts "Tipo Casilla Calle,  Titulo Propiedad:  "+@titulo_propiedad.to_s  
    end

   

  
  end
end

File: PRACTICAS/PRACTICA_ACTUAL/JAVA/JAVA/Civitas_P1/src/civitas_p1/CasillaCalle.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas_p1;

import java.util.ArrayList;

/**
 *
 * @author javierramirezp
 */
public class CasillaCalle extends CasillaDescanso{
    private TituloPropiedad tituloPropiedad;
    
    CasillaCalle(String nombre,  TituloPropiedad tituloPropiedad){
    
        super(nombre);
        this.tituloPropiedad=tituloPropiedad;
    
    }
    @Override
     void recibeJugador(int iactual, ArrayList<Jugador> todos){
        
        if(jugadorCorrecto(iactual, todos)){
            
            informe(iactual, todos);
                  
            if(!tituloPropiedad.tienePropietario()){
                
                todos.get(iactual).puedeComprarCasilla();
            }else{
                
                tituloPropiedad.tramitarAlquiler(todos.get(iactual));
                      
                
            }
           
        }
       
    }
    public TituloPropiedad getTituloPropiedad(){
        return tituloPropiedad;
    }
    @Override
    public String toString(){   
        return " llamada " + nombre + " y " + tituloPropiedad.toString();
    }
}
